api/replica.go

```diff
@@ -0,0 +1,30 @@
+package api
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gorilla/mux"
+
+	"github.com/rancher/longhorn-manager/types"
+)
+
+func (s *Server) ReplicaDelete(w http.ResponseWriter, req *http.Request) {
+	name := mux.Vars(req)["name"]
+	path := types.ReplicaDirectoryOnHost + name
```

Comment:
This is wrong. A replica is associated with the specified node and specified directory, not only the default directory.

I don't think you need to add another call for just cleaning up. We can simply add a clean up function in the replica controller, and when it see the replica on it's node get deleted, the local replica controller can proceed with cleanup. In this case, no need to validate ownerID.